### PR TITLE
Close #25

### LINKED ISSUE #25: Login - check valid email address

### PR DESCRIPTION
**Pull request by tiffanyfong**
@hmc-cs-asheppard @hmc-cs-dlan @daking2014 

check for valid email in register vc. 

used class extensions (to add new functionality) and constraints (so things look good on all phone models)

### CODE REVIEW

File: PaWPal/PaWPal/RegisterViewController.swift [hunk 1]
```diff
@@ -25,18 +25,22 @@ class RegisterViewController: UIViewController {
...
         let userEmail = userEmailTextField.text
         let userPassword = userPasswordTextField.text
         let userRepeatPassword = userRepeatPasswordTextField.text
         
         // Check validity
-        // TODO: check valid email address
         if (userEmail!.isEmpty || userPassword!.isEmpty || userRepeatPassword!.isEmpty) {
             displayAlert("All fields must be filled", handler: nil)
         }
+        if (!userEmail!.isValidEmail()) {
```
Comment (daking2014) [inline PaWPal/PaWPal/RegisterViewController.swift:41]: @tiffanyfong Is there an extra exclamation mark here?

Comment (tiffanyfong) [reply PaWPal/PaWPal/RegisterViewController.swift:41]: the first ! is a NOT operator, the second one is swift "unwrapping" an optional value - basically, optional values come in Some(value) or None, and ! forcefully unwraps it (a ? might be safer... i'll play around with it). It seems to be based on Scala options, which i've used before

Comment (daking2014) [reply PaWPal/PaWPal/RegisterViewController.swift:41]: Oh ok
What makes this an optional value?

Comment (tiffanyfong) [reply PaWPal/PaWPal/RegisterViewController.swift:41]: i think that's just a swift thing. text fields are just optional values and we have to unwrap it safely. still trying to fiddle with things

**Pull request merged**